### PR TITLE
Nodejs: readRow enhancements

### LINKED ISSUE #2634: Allow missing entries when reading from rows
I assume this is considered a feature request. In `nodejs-polars`, I want to _tabulate_ a javascript an array-nesting-objects into a _polars_ dataframe.
It's a straightforward task _if_ we have a _consistent_ structure with all properties existing in all objects:
```javascript
const myData = [
    {
        "id": "a",
        "name": "fred",
        "country": "france",
        "age": 30
    },
    {
        "id": "b",
        "name": "alexandra",
        "country": "usa",
        "age": 40
    },
    {
        "id": "c",
        "name": "george",
        "country": "argentina",
        "age": 50
    }
]
```
So we can do:
```javascript
const pl = require("nodejs-polars")

const output = pl.DataFrame(myData)
```
And we get a _polars_ dataframe as expected.

**However,** a slight inconsistency in the properties (across objects) causes it to fail:
```javascript
const myDataInconsistent = [
    {
        "id": "a",
        "name": "fred",
        "country": "france",
        "age": 30,
        "city": "paris" // there's no "city" property elsewhere
    },
    {
        "id": "b",
        "name": "alexandra",
        "country": "usa",
        "age": 40
    },
    {
        "id": "c",
        "name": "george",
        "country": "argentina",
        "age": 50
    }
]

const output = pl.DataFrame(myData)
``` 
> Error: Lengths don't match: Could not create a new DataFrame from Series. The Series have different lengths

---
Ideally, I would expect that properties that don't exist for some records but do appear in others, will either be populated with the value or become `null` in the dataframe.

---
[Originally I posted this as a question on Stack Overflow.](https://stackoverflow.com/q/71100864/6105259)

### PR DESCRIPTION
**Pull request by universalmind303**
Adds some additional functionality to read rows. 
- allows missing entries in schema, closes #2634 
- adds a 'schema' parameter for specifying own schema. (this will coerce any values that do not match given schema).
- adds `inferSchemaLength` for reading from rows. 

Also adds `unique_stable`

### CODE REVIEW

File: nodejs-polars/src/conversion/from.rs [hunk 1]
```diff
@@ -116,6 +118,62 @@ impl FromJsUnknown for AnyValue<'_> {
...
+                } else {
+                    Ok(DataType::Utf8)
+                }
+            }
+            _ => panic!("not supported"),
+        }
+    }
+}
+
+fn coerce_data_type<A: Borrow<DataType>>(datatypes: &[A]) -> DataType {
```
Comment (ritchie46) [inline nodejs-polars/src/conversion/from.rs:144]: Maybe use the [https://github.com/pola-rs/polars/blob/9c82106cba1dd9f31e4e7893020dfc388742fd6b/polars/polars-core/src/utils/mod.rs#L374] for DRYness?

**Review by ritchie46 — COMMENTED** [review]
Nice, is there also a more general `infer_schema(&[Vec<DataType>]` that could be used upstream?

File: nodejs-polars/src/conversion/from.rs (hunk 1 above)
Comment (universalmind303) [reply nodejs-polars/src/conversion/from.rs:144]: Oh nice, I didn't realize that was there.

**Comment by universalmind303** [comment]
> Nice, is there also a more general `infer_schema(&[Vec<DataType>]` that could be used upstream?

Hmm, Ill see if I can create a more generic approach.

**Comment by universalmind303** [comment]
@ritchie46 I managed to get a generic `infer_schema` working, and I moved that logic into `frame/row`. The row reader for py & rust still uses the current implementation. I didn't want this to be a big refactor, but I could submit a followup PR to have py & rust use the same logic.

**Pull request merged**